google-cloud-datastore/lib/google/cloud/datastore/service.rb

```diff
@@ -140,9 +142,25 @@ def generate_read_options consistency, transaction
             return Google::Cloud::Datastore::V1::ReadOptions.new(
               transaction: transaction
             )
+          elsif read_time
+            return Google::Cloud::Datastore::V1::ReadOptions.new(
+              read_time: read_time_to_timestamp(read_time)
+            )
           end
           nil
         end
+
+        def read_time_to_timestamp time
+          return nil if time.nil?
+
+          # Force the object to be a Time object.
+          time = time.to_time.utc
+
+          Google::Protobuf::Timestamp.new(
+            seconds: time.to_i,
+            nanos:   time.usec * 1000
```

Comment:
[Nit] Don't use the tabular space alignment. Prefer the simpler version as below:

```rb
Google::Protobuf::Timestamp.new(
  seconds: time.to_i,
  nanos: time.usec * 1000
)
```

Reply:
Updated